File: poker-hands/python/tests/card_builder.py
from poker_hands import Card, Rank, Suit


class CardBuilder:
    def __init__(self) -> None:
        self._rank: Rank = Rank.TWO
        self._suit: Suit = Suit.CLUBS

    def of_rank(self, rank: Rank) -> "CardBuilder":
        self._rank = rank
        return self

    def of_suit(self, suit: Suit) -> "CardBuilder":
        self._suit = suit
        return self

    def build(self) -> Card:
        return Card(self._rank, self._suit)

File: poker-hands/typescript/tests/CardBuilder.ts
import { Card, card } from '../src/Card.js';
import { Rank } from '../src/Rank.js';
import { Suit } from '../src/Suit.js';

export class CardBuilder {
  private rank: Rank = Rank.Two;
  private suit: Suit = Suit.Clubs;

  ofRank(rank: Rank): this { this.rank = rank; return this; }
  ofSuit(suit: Suit): this { this.suit = suit; return this; }

  build(): Card { return card(this.rank, this.suit); }
}
